File: Managed Source/Arsenal.ImageMounter.Devio/Server/GenericProviders/DummyProvider.vb

''''' DummyProvider.vb
''''' 
''''' Copyright (c) 2012-2021, Arsenal Consulting, Inc. (d/b/a Arsenal Recon) <http://www.ArsenalRecon.com>
''''' This source code and API are available under the terms of the Affero General Public
''''' License v3.
'''''
''''' Please see LICENSE.txt for full license terms, including the availability of
''''' proprietary exceptions.
''''' Questions, comments, or requests for clarification: http://ArsenalRecon.com/contact/
'''''

Imports System.Runtime.InteropServices

Namespace Server.GenericProviders

    Public NotInheritable Class DummyProvider
        Implements IDevioProvider

        ''' <summary>
        ''' Event when object is about to be disposed
        ''' </summary>
        Public Event Disposing As EventHandler Implements IDevioProvider.Disposing

        ''' <summary>
        ''' Event when object has been disposed
        ''' </summary>
        Public Event Disposed As EventHandler Implements IDevioProvider.Disposed

        Public Sub New(Length As Long)

            _Length = Length

        End Sub

        Public ReadOnly Property Length As Long Implements IDevioProvider.Length

        Public ReadOnly Property SectorSize As UInteger Implements IDevioProvider.SectorSize
            Get
                Return 512
            End Get
        End Property

        Public ReadOnly Property CanWrite As Boolean Implements IDevioProvider.CanWrite
            Get
                Return True
            End Get
        End Property

        Public ReadOnly Property SupportsShared As Boolean Implements IDevioProvider.SupportsShared
            Get
                Throw New NotImplementedException()
            End Get
        End Property

        Public Sub SharedKeys(Request As IMDPROXY_SHARED_REQ, <Out> ByRef Response As IMDPROXY_SHARED_RESP, <Out> ByRef Keys() As ULong) Implements IDevioProvider.SharedKeys
            Throw New NotImplementedException()
        End Sub

        Public Function Read(buffer As IntPtr, bufferoffset As Integer, count As Integer, fileoffset As Long) As Integer Implements IDevioProvider.Read
            Throw New NotImplementedException()
        End Function

        Public Function Read(buffer() As Byte, bufferoffset As Integer, count As Integer, fileoffset As Long) As Integer Implements IDevioProvider.Read
            Throw New NotImplementedException()
        End Function

        Public Function Write(buffer As IntPtr, bufferoffset As Integer, count As Integer, fileoffset As Long) As Integer Implements IDevioProvider.Write
            Throw New NotImplementedException()
        End Function

        Public Function Write(buffer() As Byte, bufferoffset As Integer, count As Integer, fileoffset As Long) As Integer Implements IDevioProvider.Write
            Throw New NotImplementedException()
        End Function

        Public Sub Dispose() Implements IDisposable.Dispose
        End Sub

    End Class

End Namespace

File: Managed Source/Arsenal.ImageMounter/Devio/Server/GenericProviders/DummyProvider.cs
//  
//  Copyright (c) 2012-2025, Arsenal Consulting, Inc. (d/b/a Arsenal Recon) <http://www.ArsenalRecon.com>
//  This source code and API are available under the terms of the Affero General Public
//  License v3.
// 
//  Please see LICENSE.txt for full license terms, including the availability of
//  proprietary exceptions.
//  Questions, comments, or requests for clarification: http://ArsenalRecon.com/contact/
// 

using System;
using System.Threading;
using System.Threading.Tasks;

namespace Arsenal.ImageMounter.Devio.Server.GenericProviders;

public sealed class DummyProvider(long Length) : IDevioProvider
{
    /// <summary>
    /// Event when object is about to be disposed
    /// </summary>
    public event EventHandler? Disposing;

    /// <summary>
    /// Event when object has been disposed
    /// </summary>
    public event EventHandler? Disposed;

    public long Length { get; } = Length;

    public uint SectorSize => 512U;

    public bool CanWrite => true;

    public bool SupportsParallel => true;

    bool IDevioProvider.SupportsShared => false;

    bool IDevioProvider.ForceSingleThread { get; set; }

    void IDevioProvider.SharedKeys(IMDPROXY_SHARED_REQ Request, out IMDPROXY_SHARED_RESP Response, out ulong[] Keys) => throw new NotImplementedException();

    int IDevioProvider.Read(nint buffer, int bufferoffset, int count, long fileoffset) => throw new NotImplementedException();

    int IDevioProvider.Read(byte[] buffer, int bufferoffset, int count, long fileoffset) => throw new NotImplementedException();

    int IDevioProvider.Read(Span<byte> buffer, long fileoffset) => throw new NotImplementedException();

    ValueTask<int> IDevioProvider.ReadAsync(Memory<byte> buffer, long fileoffset, CancellationToken cancellationToken) => throw new NotImplementedException();

    int IDevioProvider.Write(nint buffer, int bufferoffset, int count, long fileoffset) => throw new NotImplementedException();

    int IDevioProvider.Write(byte[] buffer, int bufferoffset, int count, long fileoffset) => throw new NotImplementedException();

    int IDevioProvider.Write(ReadOnlySpan<byte> buffer, long fileoffset) => throw new NotImplementedException();

    ValueTask<int> IDevioProvider.WriteAsync(ReadOnlyMemory<byte> buffer, long fileoffset, CancellationToken cancellationToken) => throw new NotImplementedException();

    public void Dispose()
    {
        Disposing?.Invoke(this, EventArgs.Empty);
        Disposed?.Invoke(this, EventArgs.Empty);
    }
}

